draft-kazuho-httpbis-priority.md

```diff
@@ -112,33 +112,93 @@ Unknown parameters MUST be ignored.
 
 ## urgency
 
-The `urgency` parameter takes one of the following sh-tokens as the value that
-indicates how an HTTP response affects the usage of other responses:
+The `urgency` parameter takes an integer between -1 and 6 as shown below:
 
-* `blocking` indicates that the response prevents other responses from being
-  used.
-* `document` indicates that the response contains the document that is being
-  processed.
-* `non-blocking` indicates that the response does not prevent the client from
-  using the document even though the response is being used or referred to by
-  the document.
+| Urgency         | Definition                      |
+|----------------:|:--------------------------------|
+|              -1 | blocking ({{blocking}})         |
+|               0 | document ({{document}})         |
+| between 1 and 5 | non-blocking ({{non-blocking}}) |
+|               6 | background ({{background}})     |
+{: #urgencies title="Urgencies"}
 
-The default value is `document`.
+The value is encoded as an sh-integer.  The default value is zero.
 
-A server SHOULD transmit HTTP responses in the order of their urgency:
-`blocking` first, followed by `document`, followed by `non-blocking`.
+A server SHOULD transmit HTTP responses in the order of their urgency values.
+The lesser the value is, the higher the precedence is.
 
 The following example shows a request for a CSS file with the urgency set to
-`blocking`:
+`-1`:
 
 ~~~ example
 :method = GET
 :scheme = https
 :authority = example.net
 :path = /style.css
-priority = urgency=blocking
+priority = urgency=-1
 ~~~
 
+The definition of the urgencies and their expected use-case are described below.
+Endpoints SHOULD respect the definition of the values when assigning urgencies.
+
+### blocking
+
+The blocking urgency indicates that the response prevents other responses from
+being used.
+
+Value of -1 is used to represent this urgency.
+
+For example, a stylesheet being referred from the HEAD section of an HTML
+document blocks a web browser from rendering the HTML.  In such case, the
```

Comment:
> from rendering the HTML

from rendering the remainder of the HTML

Reply:
Doesn't the CSS, by default, block the first paint?

I could well be wrong, but to me it seems that https://developers.google.com/web/fundamentals/performance/critical-rendering-path/render-blocking-css states as such.

Reply:
Yes, I was conflating some things here, good point. CSS in the HEAD indeed blocks the rendering. JS in the HEAD also blocks parsing of the remainder of the HTML, which was probably what I was getting at here. 

However, CSS in the BODY blocks the rendering of the remainder (see https://jakearchibald.com/2016/link-in-body/). 

If the example indeed just mentions CSS in the HEAD, then the original text was correct.

Reply:
Thanks. Does ebdf06a make things clearer? Please take a look.

Reply:
I think the changes in https://github.com/kazuho/draft-kazuho-httpbis-priority/commit/ebdf06ac255c5cb79b4db6f4fba4da20d2638e77 are (partially) incorrect or can be read that way.

CSS linked in the body will also block rendering, it's not just CSS in the head. The change can be read as though only CSS in the head will block rendering.

I feel for this one we should just stick to your original text, which was correct, just not exhaustive (though, as I've said in other places, I don't think we need to be that exhaustive in the text).

Reply:
Thank you for your thoughts. Yeah it's hard to strike the best balance between conciseness and clearness.

For this, I think we can simplify. Went with 99eeef4.

Reply:
Nice and clear :)